pkg/protocol/engine/booker/inmemorybooker/booker.go

```diff
@@ -173,6 +192,39 @@ func (b *Booker) markInvalid(block *blocks.Block, err error) {
 	}
 }
 
+func (b *Booker) inheritConflicts(block *blocks.Block) (conflictIDs *advancedset.AdvancedSet[iotago.TransactionID], err error) {
+	conflictIDsToInherit := advancedset.New[iotago.TransactionID]()
+
+	for _, parent := range block.ModelBlock().ParentsWithType() {
+		parentBlock, exists := b.blockCache.Block(parent.ID)
+		if !exists {
+			return nil, errors.Errorf("parent %s does not exist", parent.ID)
+		}
+
+		switch parent.Type {
+		case model.StrongParentType:
+			conflictIDsToInherit.AddAll(parentBlock.ConflictIDs())
+		case model.WeakParentType:
+			conflictIDsToInherit.AddAll(parentBlock.PayloadConflictIDs())
+		case model.ShallowLikeParentType:
+			// TODO: check whether it contains a TX, otherwise shallow like reference is invalid?
+
+			conflictIDsToInherit.AddAll(parentBlock.PayloadConflictIDs())
+			//  remove all conflicting conflicts from conflictIDsToInherit
+			for _, conflictID := range parentBlock.PayloadConflictIDs().Slice() {
+				if conflictingConflicts, exists := b.conflictDAG.ConflictingConflicts(conflictID); exists {
+					conflictIDsToInherit.DeleteAll(conflictingConflicts)
+				}
+			}
+		}
+	}
+
+	conflictIDsToInherit.AddAll(block.PayloadConflictIDs())
```

Comment:
Add a comment to explain that this is the "forking on booking", and set it through SetPayloadConflictIDs.